### DISCUSSION TITLE
Kickstarting InnerSource and Increasing Ownership motivation with Incentives - Notes from the 8/16/2020 Community Call

### DISCUSSION THREAD

**Discussion by kschueths** [InnerSourceCommons/InnerSourceMarketing#179] [General]
Here are my notes/ blog post ish from the call:
I took them from the perspective of things we are working on in our org, but the call related all of this to contribution.

Increasing Ownership motivation with Incentives

During a recent InnerSource Commons community call, we discussed strategies that InnerSource industry experts have used to motivate teams to contribute to other teams. This also relates to incentivizing people to take ownership and maintain projects and documentation.

The goal is to eventually get maintenance behaviors on the rubric to incentivize ownership and other InnerSource practices. In the meantime, we want to motivate people to take ownership of projects and actively maintain them. Active ownership will facilitate collaboration, ensure code freshness, and keep documentation current and accurate

Creating organizational buy-in to InnerSource is critical for the success of the program. Integrating incentives with career development paths will help create long term buy in with middle management.

Ownership and maintenance will lead to increased contributions because barriers to entry are decreased. Ownership may not be directly tied to a monetary incentive or reward, but incentives are critical to the success of innersource. Giving other opportunities to developers will create opportunities.

- Giving opportunities to developers to speak at conferences
- Bug hunts - snipe hunt competitions internally
- Community meetups and social gatherings to share team success stories
- Specific training opportunities
- Show contribution to another team to show professional development before joining that team
- Badging
- InnerSource/collaboration award
- Showcase success stories

Please provide additional notes!

**Comment by clcoffey** [comment]
Some of my notes on the day:

When starting an InnerSource Programme 
- Everyone in the organisation needs to understand the rationale and buy into it because of its importance to the organisation. 
- Being collaborative and transparent at the leadership and contributor levels is essential.
- Finding ways to motivate at the leadership level is also important. 
- Having a common vocabulary and educating people around InnerSource terms
- Orchestrating some first wins - there are projects that InnerSource is easier to implement in, such as DevOps 
- Developing specific roles for DevRel to help and celebrate the contributions
- Considering and working on the discoverability of projects is important for the success of an InnerSource programme too
- Sell the idea of InnerSource to the people responsible for tracking and fixing impediments in the organisation  - use InnerSource to solve the impediments
- Consider capturing people as they come into the organisation. It can be an effective way to shift the culture. 

Motivating people
- there is no one motivation that fits all, companies need to be creative and design 3 or 4 different types of programmes 
- consider programmes that fit middle management as well as end contributors. Middle management tends to create the most amount of friction.
- There is a gap between what middle management thinks developers want and what developers actually care about. Opening that conversation before designing the incentive programmes is always a good idea. 

Tracking and measuring contributions
- It is difficult to measure the way people use the tooling
- Contributions and Pull Requests can be tracked but it is more difficult to measure their value and impact.
- can’t omit the people that help beyond the code contribution.
- Considering the C-level business goals when deciding on measurements. This will allow you to report back on what they are looking for and reinforce the benefits of the InnerSource way of working.
- Developer Velocity is usually a business goal that is hard to measure. 
- McKinsey report on Developer Velocity citing "open source culture or InnerSource" as a factor of high performing organisations: https://www.mckinsey.com/industries/technology-media-and-telecommunications/our-insights/developer-velocity-how-software-excellence-fuels-business-performance


Some ideas around measuring:
- ask people (project managers) about the impact of the contributions 
- putting a value on the reused projects (e.g.: how much would it cost to develop this from scratch?) - both the consuming team and the owner team will receive that value. At the end of the year, you will have the amount of money the company saved by reusing code..
- create an award dedicated to InnerSource and collaboration - people need to apply for it and vote in order to select the winner.
